zeppelin-zengine/src/main/java/org/apache/zeppelin/storage/LocalConfigStorage.java

```diff
@@ -123,8 +123,9 @@ private void atomicWriteToFile(String content, File file) throws IOException {
     Path destinationFilePath = defaultFileSystem.getPath(file.getCanonicalPath());
     try {
       file.getParentFile().mkdirs();
-      Files.move(tempFile.toPath(), destinationFilePath,
-              StandardCopyOption.ATOMIC_MOVE);
+      Path tempDestinationFile = defaultFileSystem.getPath(file.getCanonicalPath() + ".tmp");
+      Files.move(tempFile.toPath(), tempDestinationFile);
```

Comment:
this extra file operation can be costly in some storage system?
also if the operation fails we need to clean up an extra file? L131 for example

Reply:
Thank you for review @felixcheung 
I've pushed small improvements after your comment, although you can ignore intermediate commit.
The main change that was done is to extract temporary file into the destination directory, so we always can use atomic move because the files are guaranteed on the same file system.

Or we can use intermediate version that extracts into temporary directory, and then performs 2 moves...